api/v1/intrusiondetection_types.go

```diff
@@ -27,8 +27,41 @@ type IntrusionDetectionSpec struct {
 	// Only DeepPacketInspection is supported for this spec.
 	// +optional
 	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
+
+	// AnomalyDetection provides configuration for running AnomalyDetection Component within
+	// IntrusionDetection
+	// +optional
+	AnomalyDetection AnomalyDetectionSpec `json:"anomalyDetection,omitempty"`
```

Comment:
If this is an optional field, it needs to be a pointer.

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

> Optional fields have the following properties:
> - They have the +optional comment tag in Go.
> - They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or have a built-in nil value (e.g. maps and slices).
> - The API server should allow POSTing and PUTing a resource with this field unset.

Reply:
(Although.. I can find clear examples in the Kubernetes API iteself which ignore this rule... so maybe we're OK :sweat_smile: )